### PR TITLE
[dy] Update where the `frontend_dist_base_path` is created

### PR DESCRIPTION
**Pull request by dy46**
# Description

I updated where the `frontend_dist_base_path` folder is created because it's possible that Mage won't have permission to edit the folder where the Mage package is installed. I chose to use the `get_variables_dir()` directory since I wasn't really sure where else to put it.

I also changed the `print` statements in the `server.py` file to `logger.info` since the print statements don't show up in server logs outside of the dev environment.

# How Has This Been Tested?

- [x] Tested locally with `MAGE_BASE_PATH` environment variable set
- [X] Existing unit tests pass

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation

cc: @wangxiaoyou1993

### CODE REVIEW

File: mage_ai/server/server.py [hunk 1]
```diff
@@ -113,7 +117,7 @@ def post(self, action_type):
         self.write(dict(scheduler=dict(status=scheduler_manager.get_status())))
 
 
-def replace_base_path(base_path: str) -> None:
+def replace_base_path(base_path: str) -> str:
```
Comment (wangxiaoyou1993) [inline mage_ai/server/server.py:120]: Let's add return value to the docstring

File: mage_ai/server/server.py [hunk 2]
```diff
@@ -122,7 +126,7 @@ def replace_base_path(base_path: str) -> None:
         base_path (str): The base path to replace the placeholder with.
     """
     src = os.path.join(os.path.dirname(__file__), BASE_PATH_TEMPLATE_EXPORTS_FOLDER)
-    dst = os.path.join(os.path.dirname(__file__), BASE_PATH_EXPORTS_FOLDER)
+    dst = os.path.join(get_variables_dir(), BASE_PATH_EXPORTS_FOLDER)
```
Comment (wangxiaoyou1993) [inline mage_ai/server/server.py:129]: This is a corner case but I want to make sure we catch this.
The variables_dir could be a s3 path.

**Review by wangxiaoyou1993 — COMMENTED** [review]
Could you list the detailed test cases in the Test section?

**Comment by dy46** [comment]
@wangxiaoyou1993 I added some logic to handle the s3 case.
